multiapps-controller-client/src/main/java/org/cloudfoundry/multiapps/controller/client/ResilientCloudControllerClient.java

```diff
@@ -395,8 +397,11 @@ public void updateServiceSyslogDrainUrl(String serviceName, String syslogDrainUr
     }
 
     @Override
-    public CloudPackage asyncUploadApplication(String applicationName, Path file, UploadStatusCallback callback) {
-        return executeWithRetry(() -> delegate.asyncUploadApplication(applicationName, file, callback));
+    public CloudPackage asyncUploadApplication(String applicationName, Path file, UploadStatusCallback callback, Duration uploadTimeout) {
```

Comment:
maybe we could just split this in two methods with a more clear way to see we're calling the exponentialRetry ulpoad method

Reply:
done